### PR TITLE
[12.0][FIX] website_sale_product_minimal_price: Take into account all pricelist item

### PR DESCRIPTION
**Pull request by CarlosRoca13**
Change the get_combination_info_pricelist_atributes method to get the
correct scale of prices.

cc @Tecnativa TT29708

please @pedrobaeza @chienandalu review it

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This is not `[IMP]`, but `[FIX]`, as it wasn't working previously. The commit message can be: 

```
[FIX] website_sale_product_minimal_price: Take into account all pricelist items

Previous approach only considered pricelist items in the same pricelist as the current 
one, but you can have some items that point to another pricelists, so we are changing
the approach to get all possible items, no matter the pricelist, get the minimal quantity
which they apply to, and then get the price for that quantity to see if with the price
computation, they return a different value, marking them in the scale table. With this,
we get a compromise between performance and accuracy.

NOTE: There are still left intentionally items for "All products" that can have other
scaled prices, but as this can have a lot of impact in performance and it's improbable
that someone configures pricelists this way, it's better to not include them.
```

**Review by pedrobaeza — APPROVED** [review]
Code and functional review, taking into account my commit message comment

**Comment by CarlosRoca13** [comment]
@pedrobaeza changed

**Comment by pedrobaeza** [comment]
Don't put `@Tecnativa` on the commit message. You can put the TT, but not the mention.

**Comment by CarlosRoca13** [comment]
@pedrobaeza changed

File: website_sale_product_minimal_price/controllers/main.py [hunk 1]
```diff
@@ -16,48 +16,33 @@ def get_combination_info_pricelist_atributes(
...
+        # price scale.
+        qty_list = request.env["product.pricelist.item"].search([
+            "|",
+            ("product_id", "=", product.id),
+            "|",
+            ("product_tmpl_id", "=", product.product_tmpl_id.id),
+            ("categ_id", "in", list(map(int, product.categ_id.parent_path.split('/')[0:-1]))),
+            ("min_quantity", ">", 0),
+        ]).mapped("min_quantity")
+        qty_list = sorted(set(qty_list))
```
Comment (chienandalu) [inline website_sale_product_minimal_price/controllers/main.py:32]: How do you know that the pricelist is chained to the used one?

Comment (CarlosRoca13) [reply website_sale_product_minimal_price/controllers/main.py:32]: In this case we take all the pricelist items asociated to this product to know what are the `minimal_qty`s that have to be checked, so we will know the scaled prices. This method is used just on the grid product price scale

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@chienandalu can you confirm?

**Review by chienandalu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@CarlosRoca13 please take note for fw-porting the patch to v13 when possible.

**Pull request merged**